lib/Command/ResetDocument.php

```diff
@@ -56,30 +63,44 @@ protected function configure(): void {
 		;
 	}
 
-	/**
-	 * @param InputInterface $input
-	 * @param OutputInterface $output
-	 * @return int
-	 */
 	protected function execute(InputInterface $input, OutputInterface $output): int {
 		$fileId = $input->getArgument('file-id');
+		$all = $input->getOption('all');
 		$fullReset = $input->getOption('force');
 
-		if ($fullReset) {
-			$output->writeln('Force-reset the document session for file ' . $fileId);
-			$this->documentService->resetDocument($fileId, true);
+		if (!$fileId && !$all) {
+			$output->writeln('<error>Either --all option or file-id argument is required.</error>');
+			return 1;
+		}
+		if ($fileId && $all) {
+			$output->writeln('<error>The --all option and file id argument are exclusionary.</error>');
+			return 1;
+		}
 
-			return 0;
+		if ($all) {
+			$fileIds = array_map(static function (Document $document) {
+				return $document->getId();
+			}, $this->documentService->getAll());
+		} else {
+			$fileIds = [$fileId];
 		}
 
-		$output->writeln('Reset the document session for file ' . $fileId);
-		try {
-			$this->documentService->resetDocument($fileId);
-		} catch (DocumentHasUnsavedChangesException) {
-			$output->writeln('Not resetting due to unsaved changes');
-			return 1;
+		$rc = 0;
+		foreach ($fileIds as $id) {
+			if ($fullReset) {
+				$output->writeln('Force-reset the document session for file ' . $id);
+				$this->documentService->resetDocument($id, true);
```

Comment:
I guess we can skip the rest with continue?

Reply:
Indeed, good catch :fishing_pole_and_fish: